### PR TITLE
ci: lint commit message and PR titles

### PR DESCRIPTION
**Pull request by kwajiehao**
This PR introduces linting for commit message titles and PR titles. We lint commit message titles using the `commitlint` [library](https://github.com/conventional-changelog/commitlint) (enforced by husky hooks), and we lint PR titles using the [amannn/action-semantic-pull-request](https://github.com/amannn/action-semantic-pull-request) GitHub Action.

This is the first step of a proposal to automate our release processes using the `semantic-release` [library](https://github.com/semantic-release/semantic-release) - for more information, refer to the proposal ([link](https://docs.google.com/document/d/1XhFHl2fZAy69Jb9Ps3XXnBSsPxLScaTpsknoQ8Ei_uE/edit#heading=h.q7i73gfj0g2d)).  `semantic-release` uses semantic commit formatting to determine the next release version. This makes linting PR titles extremely important, since we use a squash-and-merge strategy for our feature branches, and the PR titles eventually become commit messages.

Note: amannn/action-semantic-pull-request requires a `GH_TOKEN` access token to run. I have used my own access token for now, but this should ideally be replaced with a team token.

### CODE REVIEW

**Review by lamkeewei — COMMENTED** [review]
Might be good to setup a separate GitHub user for Isomer CI and use a minimally scoped access token (no access to live site repos, etc) for our CI needs.

File: .github/workflows/lint-pr.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+jobs:
+  lint-pr:
+    name: Lint PR titles to ensure that they follow Conventional Commit formatting
+    runs-on: ubuntu-18.04
+    steps:
+      # Please look up the latest version from
+      # https://github.com/amannn/action-semantic-pull-request/releases
+      - name: semantic-pull-request
+        uses: amannn/action-semantic-pull-request@v3.4.0
+        if: ${{ env.DEVELOP_BRANCH == env.TARGET_BRANCH }}
```
[the remark this answers is not in the document]
Comment (lamkeewei) [reply .github/workflows/lint-pr.yml:20]: Ok actually ignore the last comment. I guess this check only make sense for PRs that are going to be squashed and merged into `develop`.

**Comment by kishore03109** [comment]
@seaerchin @alexanderleegs shall we close this? haha we already use commitizen anyways

**Comment by QiluXie** [comment]
> @seaerchin @alexanderleegs shall we close this? haha we already use commitizen anyways

Close it

**Pull request closed without merging**